### DISCUSSION TITLE
Singularity Hub?

### DISCUSSION THREAD

**Discussion by davidfrantz** [davidfrantz/force#141] [Ideas]
I just saw @thielfab using FORCE with ``Singularity``.

Is there something like ``Docker Hub`` for ``Singularity``?

**Comment by thielfab** [comment]
Not that I know of. Here at UG we have a self-hosted version of Harbor https://goharbor.io/
And prior to using your pre-built docker images, my workflow was to 

- build an image locally using docker
- deploy it to the registry (harbor)
- on the server: pull image using singularity
- run image using singularity (no docker installed on server!)

**Comment by williww** [comment]
There is kind of hub for Singularity: **https://cloud.sylabs.io/library**
Not comparable to hub.docker.com, but ok for building FORCE-containers.
Used their ubuntu:20.4 image to build FORCE. @davidfrantz: will discuss this next days, perhaps ;-)

Personally, i prefer *Singularity* for some reason. Keywords: root, /var/log/docker...

**Reply by davidfrantz** [reply]
Interesting!

I didn't find anything on the usage conditions, though.

Is it free? Are there quotas for storing, pushing and pulling?

**Comment by williww** [comment]
How to push (publish?) your (SIF) container?

Read this: https://sylabs.io/guides/latest/user-guide/cloud_library.html

I think (hope), Terms and Conditions will be explained/listed during account or token creation.

@davidfrantz : Currently i am testing some updated Singularity-SIF (FORCE). DEF-Files (you know "Dockerfile") are available. Might be pushed (GitHub)? My Repo here? Yours? Lets talk internaly...
